peer/peer.go

```diff
@@ -433,6 +433,11 @@ type Peer struct {
 	cfg     Config
 	inbound bool
 
+	// These fields track the peer's whitelisted status and ban score.
+	isWhitelisted bool
+	banScore      *DynamicBanScore
+	banManager    BanManager
```

Comment:
Similarly, this doesn't make sense to me. The ban manager, as it name implies, is the thing that manages peers, not the other way around.